tools/third_party/mozlog/mozlog/handlers/valgrindhandler.py

```diff
@@ -0,0 +1,138 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import re
+
+from .base import BaseHandler
+
+
+class ValgrindHandler(BaseHandler):
+    def __init__(self, inner):
+        BaseHandler.__init__(self, inner)
+        self.inner = inner
+        self.vFilter = ValgrindFilter()
+
+    def __call__(self, data):
+        tmp = self.vFilter(data)
+        if tmp is not None:
+            self.inner(tmp)
+
+
+class ValgrindFilter(object):
+    """
+    A class for handling Valgrind output.
+
+    Valgrind errors look like this:
+
+    ==60741== 40 (24 direct, 16 indirect) bytes in 1 blocks are definitely lost in loss
+                  record 2,746 of 5,235
+    ==60741==    at 0x4C26B43: calloc (vg_replace_malloc.c:593)
+    ==60741==    by 0x63AEF65: PR_Calloc (prmem.c:443)
+    ==60741==    by 0x69F236E: PORT_ZAlloc_Util (secport.c:117)
+    ==60741==    by 0x69F1336: SECITEM_AllocItem_Util (secitem.c:28)
+    ==60741==    by 0xA04280B: ffi_call_unix64 (in /builds/s1ave/m-in-l64-valgrind-000000000000/objdir/toolkit/library/libxul.so) # noqa
```

Comment:
@jgraham I changed "slave" to "s1ave" here. Such word is not allowed in chromium. Do you have a suggestion what to do here?

Reply:
I think you could replace it with `worker`; it's only example output so I don't think that will break anything, and actually matches what gecko CI uses these days.